cypress/utils/setup-utils.js

```diff
@@ -70,13 +77,182 @@ export const cleanUpRunOktaOrgs = (specRunVersionName) => {
   });
 };
 
-export const setupRunData = (specRunVersionName) => {
+export const setupOrgFacility = (specRunVersionName) => {
   let orgName = createOrgName(specRunVersionName);
   let facilityName = createFacilityName(specRunVersionName);
-  createAndVerifyOrganization(orgName)
+  return createAndVerifyOrganization(orgName)
     .then(() => getOrganizationsByName(orgName))
     .then((res) =>
       accessOrganization(res.body.data.organizationsByName[0].externalId),
     )
     .then(() => addMockFacility(facilityName));
 };
+
+export const getCreatedFacility = (specRunVersionName) => {
+  let orgName = createOrgName(specRunVersionName);
+  return getOrganizationsByName(orgName).then((res) => {
+    let orgs = res.body.data.organizationsByName;
+    let org = orgs.length > 0 ? orgs[0] : null;
+    if (org) {
+      let facilities = org.facilities;
+      return facilities.length > 0 ? facilities[0] : null;
+    }
+  });
+};
+
+export const setupPatient = (specRunVersionName, patient) => {
+  return getCreatedFacility(specRunVersionName).then((facility) => {
+    const addPatientVariables = {
+      facilityId: facility.id,
+      firstName: patient.firstName,
+      lastName: patient.lastName,
+      birthDate: patient.dobForInput,
+      street: patient.address,
+      city: patient.city,
+      state: patient.state,
+      country: "USA",
+      zipCode: patient.zip,
+      telephone: patient.phone,
+    };
+    return addPatient(addPatientVariables).then((result) => {
+      return result.body.data.addPatient;
+    });
+  });
+};
+
+export const setupCovidOnlyDevice = (specRunVersionName, covidOnlyDevice) => {
+  let specimenTypeId,
+    supportedDiseaseId,
+    createdDeviceId,
+    createDeviceTypeVariables;
+
+  return getSpecimenTypes()
+    .then((result) => {
+      const specimenTypes = result.body.data.specimenTypes;
+      specimenTypeId =
+        specimenTypes.length > 0 ? specimenTypes[0].internalId : null;
+    })
+    .then(() => getSupportedDiseases())
+    .then((result) => {
+      const supportedDiseases = result.body.data.supportedDiseases;
+      supportedDiseaseId =
+        supportedDiseases.length > 0 ? supportedDiseases[0].internalId : null;
+      createDeviceTypeVariables = {
```

Comment:
suggestion: I think we could simply / dedupe the logic here and in the next function for multiplex device creation, since there only seems to be minor differences between covid and multiplex devices. I'm envisioning three functions: a central generic function that just does the device -> facility mapping and then two wrappers (one for COVID and one for Multiplex) that differentiates between the two cases.

In the COVID / multiplex wrappers (roughly everything from here in the then chain upwards), we could do the data preparation for different device types. Those wrappers could then call a generic function that maps that device data onto a facility (everything from here down) and not know anything about the contents of the device data.

Let me know if that makes sense / if a more high-fidelity example would be helpful

Reply:
Just pushed an abstraction for this, let me know what you think when you get a chance to re-review

Reply:
looks great, thanks!